File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
	int hIndex;   
	
    public int solution(int[] citations) {
    	Arrays.sort(citations);

    	int curr =(int) Math.ceil((double)citations.length/2)-1;
    	hIndex = citations[curr]; 
    	
    	while(hIndex>0) {
    		long upperLine = Arrays.stream(citations).filter(x->x>=hIndex).count();
    		long underLine = Arrays.stream(citations).filter(x->x<=hIndex).count();
    		
    		if(hIndex<=upperLine && hIndex>=underLine) {
    			break;
    		}
    		
    		hIndex--;
    	}

        return hIndex;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    hIndex = len(citations)
    
    while hIndex > 0 :
        uCnt = 0
        lCnt = 0
        
        for c in citations:
            if c >= hIndex:
                uCnt= uCnt +1
            if c <= hIndex:
                lCnt = lCnt + 1
                
        if hIndex <= uCnt and hIndex >= lCnt :
            break
            
        hIndex = hIndex - 1
    
    return hIndex